flipt-client-go/config.go

```diff
@@ -91,6 +92,40 @@ func WithRequestTimeout(timeout time.Duration) Option {
 	}
 }
 
+// WithTLSConfig sets the TLS configuration for the client using the standard library tls.Config.
+// This provides maximum flexibility for configuring TLS settings including custom CAs,
+// mutual TLS authentication, and certificate verification options.
+func WithTLSConfig(tlsConfig *tls.Config) Option {
```

Comment:
The order of options can lead to unexpected behavior. For example, using `WithTLSConfig` followed by `WithHTTPClient`  may not work as intended.
While this combination is probably rare in real world, I’d suggest mentioning the order dependency in the Go doc to avoid confusion.

Reply:
good point. will update the godocs and readme to explain this